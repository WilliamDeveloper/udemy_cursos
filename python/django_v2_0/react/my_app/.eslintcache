[{"D:\\_dev_\\work\\github\\udemy_cursos\\react\\my_app\\src\\reportWebVitals.js":"1","D:\\_dev_\\work\\github\\udemy_cursos\\react\\my_app\\src\\components\\Autores.js":"2"},{"size":362,"mtime":1607199454780,"results":"3","hashOfConfig":"4"},{"size":1141,"mtime":1572496259000,"results":"5","hashOfConfig":"4"},{"filePath":"6","messages":"7","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1axvtxd",{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"D:\\_dev_\\work\\github\\udemy_cursos\\react\\my_app\\src\\reportWebVitals.js",[],"D:\\_dev_\\work\\github\\udemy_cursos\\react\\my_app\\src\\components\\Autores.js",[]]