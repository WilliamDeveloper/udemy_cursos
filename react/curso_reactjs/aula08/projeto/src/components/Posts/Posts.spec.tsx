import {render, screen} from "@testing-library/react";
import {Posts} from ".";

import postsPropsMock from './mock'

const props = postsPropsMock

describe('<Posts />',()=>{

    it('should render Posts',()=>{

        // @ts-ignore
        render(<Posts {...props} />)
        expect(screen.getAllByRole('heading',{name: /title/i})).toHaveLength(3)
        expect(screen.getAllByRole('img',{name: /title/i})).toHaveLength(3)
        expect(screen.getAllByText(/body/i)).toHaveLength(3)

    })

    it('should match snapshot ',()=>{
        const {container} = render(<Posts {...props} />)
        expect(container.firstChild).toMatchSnapshot()

    })

})
