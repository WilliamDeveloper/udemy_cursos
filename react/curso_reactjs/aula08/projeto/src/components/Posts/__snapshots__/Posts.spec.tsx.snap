// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot  1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      src="https://via.placeholder.com/600/771796"
      title="title 1"
    />
    <div
      class="post-content"
    >
      <h2>
        title 1
        -
        1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      src="https://via.placeholder.com/600/771796"
      title="title 2"
    />
    <div
      class="post-content"
    >
      <h2>
        title 2
        -
        2
      </h2>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      src="https://via.placeholder.com/600/771796"
      title="title 3"
    />
    <div
      class="post-content"
    >
      <h2>
        title 3
        -
        3
      </h2>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
